### PR TITLE
Implement the user dashboard with calls to server to pull data.

### PR DESCRIPTION
**Pull request by sscchan**
Also added mechanisms for login and logout.

### CODE REVIEW

File: client/index.jsx [hunk 1]
```diff
@@ -14,47 +14,97 @@ const UserLegislatorsInfo = require(__dirname + '/src/components/UserLegislators
...
+
+  // Need check with the server to see if user is autheticated
+  componentDidMount() {
+    $.get('login')
+      .done(function(data) {
+        this.setState({        
+          isVerifyingUserSession: false,
+          isUserLoggedIn: true,
+          username: data.username,
+          userLocation: data.location
```
Comment (benzeg) [inline client/index.jsx:42]: data.location points to user full address
data.geoLocation points to user coordinates
I believe this is a necessary distinction for when we want to display the user's address in client

File: client/src/components/UserDashBoard.jsx [hunk 2]
```diff
@@ -1,16 +1,121 @@
...
+
+    this.handleAddMonitoredWords = this.handleAddMonitoredWords.bind(this);
+    this.handleRemoveMonitoredWords = this.handleRemoveMonitoredWords.bind(this);
+  }
+  componentDidMount() { 
+    // Retrieve user monitored keywords and associated bills to populate the dashboard
+    $.get('user/' + this.props.username + '/keywords')
+      .done(data => this.setState({
+        isFetching: false,
+        monitoringResults: data,
```
Comment (benzeg) [inline client/src/components/UserDashBoard.jsx:23]: data will be an object with the following format:
{keywords: {"keyword1": {keyword: "keyword1",
                                        associatedKeywords: [...,...,],
                                        bills: [id1, id2, id3]}
                  }

File: client/src/components/UserDashBoard.jsx [hunk 3]
```diff
@@ -1,16 +1,121 @@
...
+      context: this,
+      data: {
+        keyword: words
+      },
+      dataType: 'json'
+    })
+      .done(function(data) {
+
+        let revisedMonitoringResults = this.state.monitoringResults.slice();
+        revisedMonitoringResults.unshift(data);
```
Comment (benzeg) [inline client/src/components/UserDashBoard.jsx:46]: data will be an object with the following format:
{keywords: {"keyword1": {keyword: "keyword1",
associatedKeywords: [...,...,],
bills: [id1, id2, id3]}
}

File: client/src/components/UserDashBoard.jsx [hunk 4]
```diff
@@ -1,16 +1,121 @@
...
+          this.setState({
+            monitoringResults: revisedMonitoringResults
+          });
+        }
+      });
+  }
+
+  handleRemoveMonitoredWords(words) {
+    //DELETE message to server to remove a monitored word(s) string
+    $.ajax('user/' + this.props.username + '/keywords', {
```
Comment (benzeg) [inline client/src/components/UserDashBoard.jsx:61]: Here we can also send you an updated data with the following data type:

{keywords: {"keyword1": {keyword: "keyword1",
associatedKeywords: [...,...,],
bills: [id1, id2, id3]}
}

**Review by benzeg — CHANGES_REQUESTED** [review]
commented on conflicts in user data types

File: client/src/components/UserDashBoard.jsx (hunk 4 above)
Comment (sscchan) [reply client/src/components/UserDashBoard.jsx:61]: can you explain the rationale behind this particular deviation from the agreed format?
shouldn't it be [{word: '<monitoredwords>', associatedKeywords: [...,...,...], relatedBills['id1','id2'}, {word: '<monitoredwords2>, associatedKeywords: [...,...,...], relatedBills['id3','id5'}, {word: ] as per the whiteboarding session

File: client/src/components/UserDashBoard.jsx (hunk 3 above)
Comment (sscchan) [reply client/src/components/UserDashBoard.jsx:46]: can you explain the rationale behind this particular deviation from the agreed format?
shouldn't it be [{word: '<monitoredwords>', associatedKeywords: [...,...,...], relatedBills['id1','id2'}, {word: '<monitoredwords2>, associatedKeywords: [...,...,...], relatedBills['id3','id5'}, {word: ] as per the whiteboarding session

File: client/src/components/UserDashBoard.jsx (hunk 2 above)
Comment (sscchan) [reply client/src/components/UserDashBoard.jsx:23]: can you explain the rationale behind this particular deviation from the agreed format?
shouldn't it be [{word: '<monitoredwords>', associatedKeywords: [...,...,...], relatedBills['id1','id2'}, {word: '<monitoredwords2>, associatedKeywords: [...,...,...], relatedBills['id3','id5'}, {word: ] as per the whiteboarding session

File: client/index.jsx (hunk 1 above)
Comment (sscchan) [reply client/index.jsx:42]: my understand of our whiteboarding is that the user lat / long from the GET is in the format {location: {lat: <lat>, long: <long>}} and 
address (via POST in /signup) has the form {address: {houseNum: <housenumber>, streetName: <streetName>, city: <city>, state <state>}}

Is there a compelling reason to change the previously agreed upon interface?

Comment (benzeg) [reply client/index.jsx:42]: Full address is listed under 'location' in our DB schema and lat+long are separate fields, so I kept location=full address and created 'geoLocation' for lat+long

File: client/src/components/UserDashBoard.jsx (hunk 2 above)
Comment (benzeg) [reply client/src/components/UserDashBoard.jsx:23]: Keeping word objects in a list object allows us to easily refer to a word for deletion.

File: client/src/components/UserDashBoard.jsx (hunk 3 above)
Comment (benzeg) [reply client/src/components/UserDashBoard.jsx:46]: Keeping word objects in a list object allows us to easily refer to a word for deletion.

File: client/src/components/UserDashBoard.jsx (hunk 4 above)
Comment (benzeg) [reply client/src/components/UserDashBoard.jsx:61]: Keeping word objects in a list object allows us to easily refer to a word for deletion.

File: client/src/components/UserDashBoard.jsx (hunk 2 above)
Comment (benzeg) [reply client/src/components/UserDashBoard.jsx:23]: We'll convert object into array. Keep code as is.

File: client/src/components/UserDashBoard.jsx (hunk 3 above)
Comment (benzeg) [reply client/src/components/UserDashBoard.jsx:46]: We'll convert object into array. Keep code as is.

File: client/src/components/UserDashBoard.jsx (hunk 4 above)
Comment (benzeg) [reply client/src/components/UserDashBoard.jsx:61]: We'll convert object into array. Keep code as is.

File: client/index.jsx (hunk 1 above)
Comment (sscchan) [reply client/index.jsx:42]: ok, can you define exactly the information in the body I will receive from in the response?

**Pull request merged**